From 22e540a225da939e9de02912d638402d2353b129 Mon Sep 17 00:00:00 2001
From: Jim Ramsay <i.am@jimramsay.com>
Date: Wed, 28 Apr 2010 10:46:15 -0400
Subject: [PATCH 14/14] Bug 594153 (3/3) - Allow setting alarms on any meeting

By tying the attachment view "editable" action group to comp-editor
"individual" group, these add/remove actions are marked sensitive /
insensitive in sync with the existing "Insert" menu items which are
already in the "individual" group.

Though not introduced with the other symptoms of this bug
(0597b877c5bf4d21ac4048742ddf6b11e24877ba), the descreptency fixed
here is integrally related.
---
 calendar/gui/dialogs/comp-editor.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/calendar/gui/dialogs/comp-editor.c b/calendar/gui/dialogs/comp-editor.c
index 7d27474..10390e6 100644
--- a/calendar/gui/dialogs/comp-editor.c
+++ b/calendar/gui/dialogs/comp-editor.c
@@ -36,6 +36,7 @@
 #include <glib/gstdio.h>
 #include <gdk/gdkkeysyms.h>
 #include <e-util/e-util.h>
+#include <e-util/e-binding.h>
 #include <e-util/e-dialog-utils.h>
 #include <e-util/e-util-private.h>
 #include <e-util/gconf-bridge.h>
@@ -1553,6 +1554,7 @@ comp_editor_init (CompEditor *editor)
 	GtkTargetList *target_list;
 	GtkTargetEntry *targets;
 	GtkActionGroup *action_group;
+	GtkActionGroup *action_group_2;
 	GtkAction *action;
 	GtkWidget *container;
 	GtkWidget *widget;
@@ -1746,6 +1748,13 @@ comp_editor_init (CompEditor *editor)
 	gtk_window_set_type_hint (
 		GTK_WINDOW (editor), GDK_WINDOW_TYPE_HINT_NORMAL);
 
+	action_group = comp_editor_get_action_group (editor, "individual");
+	action_group_2 = e_attachment_view_get_action_group (view, "editable");
+
+	e_binding_new (
+		action_group, "sensitive",
+		action_group_2, "sensitive");
+
 	/* Listen for attachment store changes. */
 
 	store = e_attachment_view_get_store (view);
-- 
1.7.1

